implementations/rust/ockam/ockam_command/src/lib.rs

```diff
@@ -446,3 +448,62 @@ pub(crate) fn replace_hyphen_with_stdin(s: String) -> String {
         s
     }
 }
+
+const EXIT_OK: i32 = 0;
+const EXIT_USAGE: i32 = 2;
+
+fn show_help(help: clap::Error) {
+    paginate(&help.render()).unwrap_or_else(|_| help.print().unwrap());
+
+    let _ = std::io::stdout().lock().flush();
+    let _ = std::io::stderr().lock().flush();
+    process::exit(if help.use_stderr() {
+        EXIT_USAGE
+    } else {
+        EXIT_OK
+    });
+}
+
```

Comment:
The following two functions could be simplified. There is a library called [`which`](https://crates.io/crates/which) you could use to get the binaries paths.

```rust
fn paginate(text: &StyledStr) -> Result<()> {
    let default_pager = get_env_with_default("PAGER", "less".to_string())?;
    let mut pagers = vec![&default_pager, "more", "bat"];
    for pager in pagers.drain(..) {
        if let Ok(path) = which::which(pager) {
            let path = path
                .to_str()
                .expect(&format!("{pager}'s path is not valid UTF-8"));
            if let Ok(()) = paginate_with(path, text) {
                break;
            }
        }
    }
    Ok(())
}

fn paginate_with(pager_path: &str, text: &StyledStr) -> Result<()> {
    let mut cmd = process::Command::new(pager_path);
    // Add needed arguments to the pager command.
    match pager_path {
        "less" => {
            // No pagination if the text fits entirely into the window.
            // Using env var in case a lesser `less` poses as `less`.
            cmd.env("LESS", "-F");
        }
        _ => {}
    }
    let mut child = cmd.stdin(process::Stdio::piped()).spawn()?;
    // Write text to pager's stdin.
    {
        let mut stdin = child.stdin.take().unwrap();
        stdin.write_all(format!("{}", text).as_bytes())?;
    }
    child.wait()?;
    Ok(())
}
```

Reply:
What would that be a the simplification? Also adding another dependency.. But more importantly, I think you are misreading my code, for this breaks when `PAGER=/opt/not-on-PATH/pager`. I added a fallback to `more` with devices in mind and don't think the overkill of probing any pager on the planet is necessary.